rust/lance-table/protos/table.proto

```diff
@@ -231,156 +232,3 @@ message DeletionFile {
   uint64 num_deleted_rows = 4;
 } // DeletionFile
 
-// Metadata of one Lance file.
```

Comment:
The old `format.proto` split into `file.proto` and `table.proto`.  In the process I changed the `package` field.  This should be a non-breaking change for protobuf as long as we aren't using these messages is a protobuf `Any` field (we aren't)